runtime/lua/vim/treesitter/query.lua

```diff
@@ -6,17 +6,55 @@ local language = require'vim.treesitter.language'
 local Query = {}
 Query.__index = Query
 
+local query_file_config = {
+  ignore_patterns = {},
+  -- for caching results
+  ignored_files = {},
+}
+
 local M = {}
 
 ---@private
-local function dedupe_files(files)
+---@param lang string
+---@param file string
+---@return boolean
+local function is_file_ignored(lang, file)
+  local ignore_patterns = query_file_config.ignore_patterns[lang]
```

Comment:
if we are going to create an inner configuration platform for treesitter (rather than using e.g. Nvim editor options, `option.c`) can we at least not add random functions to `vim.treesitter` and instead have a single `vim.treesitter.config` interface or something like that?